packages/@aws-cdk/aws-iam/lib/policy-document.ts

```diff
@@ -90,12 +91,21 @@ export class AccountPrincipal extends ArnPrincipal {
  * An IAM principal that represents an AWS service (i.e. sqs.amazonaws.com).
  */
 export class ServicePrincipal extends PolicyPrincipal {
-  constructor(public readonly service: string) {
+  /**
+   * @param scope   a Construct scoping the service principal, typically the resource that will cause the service
```

Comment:
In fact we could do this at resolution time by returning a `Token` instead of a literal value. 

A `Token` will have information about the scope it's resolved in, so you don't need to add the parameter here.

Reply:
Except the region that should be used there isn't necessarily that of the region of the stack where the principal is being used...

Reply:
*sigh* our constant dilemma... @RomainMuller intuitively it seems like you are right, but can you provide an example? I am failing to think of one. Maybe the opposite is correct, a service principal by definition always has to be "local" to the stack it's defined in?

...eventually everything is going to be a construct...

Reply:
So I was thinking of a cross-region flow where the destination region has to trust the source region's service principal to assume a role, for example. However maybe that is not a thing at all?

Reply:
@rix0rrr @skinny85 any thoughts on that? I would rather we make this a token (which will also make it easier to avoid mistakes) and support some quirky cross-region service principals explicitly than make all code in the world need to pass in another argument here.

Reply:
I had considered that as well, but I think the cases in which we want automatically pinned cross-region service principals would be exceedingly rare. And in those cases, you could still explicitly specify them (or should be able to).

Reply:
+1 for the Token option and supporting cross-region service principals explicitly... unless we have an obvious use-case when you would need cross-region, but it seems like more speculation at this point, so let' not sacrifice ergonomy for that just yet.